### PR TITLE
Changing ingress from manually scaling rc to a daemon set.

### PR DESCRIPTION
**Pull request by hyperbolic2346**
**What this PR does / why we need it**:
Quality of life fix. We used to set an RC to run the nginx ingress and then scale it in the charm as workers were added. Now we just use a daemon set to make sure it is running everywhere. Using application-specific names and roles so multiple applications can have different ingress settings.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/367

**Special notes for your reviewer**:

**Release note**:

```release-note
CDK nginx ingress is now handled via a daemon set.
```

### CODE REVIEW

File: cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import shutil
 import subprocess
 import time
+import json
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: This is already imported on line 17:
```
$ flake8 kubernetes_worker.py 
kubernetes_worker.py:23:1: F811 redefinition of unused 'json' from line 17
```

Mind fixing that up? Our automation will fuss about it.

Comment (hyperbolic2346) [reply cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: Thanks for noticing that.

**Comment by Cynerva** [comment]
/approve no-issue
/lgtm

Thanks for this :+1:

**Pull request merged**